cloudant-http/src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -79,86 +84,123 @@ public HttpConnectionInterceptorContext interceptRequest(HttpConnectionIntercept
-                            if (!errorString.matches("(?siu)" +
-                                    ".*\\\"error\\\"\\s*:\\s*\\\"credentials_expired\\\".*")) {
-                                // Wasn't a credentials expired, throw exception
-                                HttpConnectionInterceptorException toThrow = new
-                                        HttpConnectionInterceptorException(errorString);
-                                // Set the flag for deserialization
-                                toThrow.deserialize = true;
-                                throw toThrow;
-                            } else {
-                                // Was expired - set boolean to renew cookie
+        // Check if this interceptor is valid before attempting any kind of renewal
+        if (shouldAttemptCookieRequest.get()) {
+
+            HttpURLConnection connection = context.connection.getConnection();
+
+            // If we got a 401 or 403 we might need to renew the cookie
+            try {
+                boolean renewCookie = false;
+                int statusCode = connection.getResponseCode();
+
+                if (statusCode == HttpURLConnection.HTTP_FORBIDDEN || statusCode ==
+                        HttpURLConnection.HTTP_UNAUTHORIZED) {
+                    InputStream errorStream = connection.getErrorStream();
+                    try {
+                        switch (statusCode) {
+                            case HttpURLConnection.HTTP_FORBIDDEN: //403
+                                //check if it was an expiry case
+                                String errorString = IOUtils.toString(errorStream, "UTF-8");
+                                // Check using a regex to avoid dependency on a JSON library.
+                                // Note (?siu) flags used for . to also match line breaks and for
+                                // unicode
+                                // case insensitivity.
+                                if (!errorString.matches("(?siu)" +
+                                        ".*\\\"error\\\"\\s*:\\s*\\\"credentials_expired\\\".*")) {
+                                    // Wasn't a credentials expired, throw exception
+                                    HttpConnectionInterceptorException toThrow = new
+                                            HttpConnectionInterceptorException(errorString);
+                                    // Set the flag for deserialization
+                                    toThrow.deserialize = true;
+                                    throw toThrow;
+                                } else {
+                                    // Was expired - set boolean to renew cookie
+                                    renewCookie = true;
+                                }
+                                break;
+                            case HttpURLConnection.HTTP_UNAUTHORIZED: //401
+                                IOUtils.toString(errorStream, "UTF-8");
```

Comment:
Yes, I think findbugs normally does, not sure why it didn't for this. In 632cc86 I have moved the string extraction higher up and added some logging for the value which might be useful anyway.